### PR TITLE
fix(invity): refactoring Invity API data load

### PR DESCRIPTION
**Pull request by LukasRada**
Improves loading data (for coinmarket) from Invity API server.

### CODE REVIEW

**Review by tomasklim — APPROVED** [review]

File: packages/suite/src/actions/wallet/coinmarket/coinmarketCommonActions.ts [hunk 1]
```diff
@@ -126,3 +132,12 @@ export const submitRequestForm = (form?: {
         envSubmitRequestForm(form.formMethod, form.formAction, form.fields);
     }
 };
+
+export const setLoading = (isLoading: boolean, lastLoadedTimestamp?: number) => (
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/coinmarket/coinmarketCommonActions.ts]: just a code hygiene nitpick
you dont have to use redux-thunk action here (you are not working with reducers/getState)
standard redux acion should be enough, no?
```
export const setLoading = (isLoading: boolean, lastLoadedTimestamp?: number) => ({
    type: COINMARKET_COMMON.SET_LOADING,
    isLoading,
    lastLoadedTimesta
});
```

File: packages/suite/src/hooks/wallet/useCoinmarket.ts [hunk 2]
```diff
@@ -18,49 +20,58 @@ export const useInvityAPI = () => {
...
+            ])
+                .then(([buyInfoResult, [exchangeInfo, exchangeCoinInfo], sellInfoResult]) => {
+                    setIsLoading(false, Date.now());
+                    saveBuyInfo(buyInfoResult);
                     saveExchangeInfo(exchangeInfo);
                     saveExchangeCoinInfo(exchangeCoinInfo);
+                    saveSellInfo(sellInfoResult);
+                })
+                .catch(() => {
+                    setIsLoading(false);
```
Comment (szymonlesisz) [inline packages/suite/src/hooks/wallet/useCoinmarket.ts]: i'm wondering on the impact of this.
example: if loading of `exchangeInfo` fails then you will not have ANY other info (sell, buy..) it this ok?

another thing is architecture of this code, should it be in hook? (i know it was like this before)
but maybe it should be moved to some middleware which is independent from the react render process (and from mount/unmount events) ?

File: packages/suite/src/actions/wallet/coinmarket/coinmarketCommonActions.ts (hunk 1 above)
Comment (LukasRada) [reply packages/suite/src/actions/wallet/coinmarket/coinmarketCommonActions.ts:136]: This piece will be obsolete after moving the code to middleware (next comment). Thanks for letting me know.

File: packages/suite/src/hooks/wallet/useCoinmarket.ts (hunk 2 above)
Comment (LukasRada) [reply packages/suite/src/hooks/wallet/useCoinmarket.ts]: Agreed, will fix.
Any chance to upgrade TS lib es2019 -> es2020?
So we can use `Promise.allSettled` for easier handling of fulfilled or rejected promises?

Comment (szymonlesisz) [reply packages/suite/src/hooks/wallet/useCoinmarket.ts]: good question, maybe with the incoming new pipeline?
but it doesn't look like it happen anytime soon... maybe leave some TODO note there

**Comment by LukasRada** [comment]
3fad1a7 fixes another bug - navigation between accounts in coin market doesn't load data for the current account, but data for the first one stays.

**Comment by tomasklim** [comment]
I don't know how to replicate the bug from the latest fix 😢 

Please rebase it to the newest development branch to make unit tests work.

Btw is this correct that I do not get sell price? The second input does nothing in Buy and Sell tab. In Exchange tab, it works.
Screenshot 2021-06-08 at 11 46 41

**Comment by LukasRada** [comment]
Ad) bug solved via 3fad1a7
Repro steps:
1. Select an account e.g. first BTC account
2. Navigate to Trade and to Buy form
3. See outgoing HTTP requests to get buy/exchange/sell data with header `X-SuiteW-Api`
4. Select a different account e.g. second BTC account
5. Navigate to Trade and to Buy form
6. See outgoing HTTP requests to get buy/exchange/sell data with header `X-SuiteW-Api`

The bug: In the sixth step, the header value was the same as for the first account (third step) or there were no requests at all because of 10min refresh interval.
Expected behavior: HTTP headers are unique per account. And data for coin market are loaded when the account has been changed.

Ad) Prices
Yes. The behavior is correct. Price is not calculated in this step for buy and sell. Whereas in exchange price calculation is 
 intention.

**Review by szymonlesisz — APPROVED** [review]
looks way better with middleware, thanks

approving but didnt test it

you may squash it now

**Pull request merged**